### PR TITLE
Can pass refactor

### PR DESCRIPTION
**Pull request by ninjanomnom**
Completely rewrites how CanPass works so that we can have core code that always runs and is able to override default behaviors in a more extendable way. Key change of note is that the mover has a proc they can override to affect behavior. This makes `UNBLOCKABLE` movement type not a lie.

~~This code should be ready but I haven't had an opportunity to test much yet. I'll leave this in draft until I feel it's been tested enough both locally and on a live server.~~

### CODE REVIEW

**Comment by kevinz000** [comment]
god-tier per usual

File: code/game/objects/structures/holosign.dm [hunk 1]
```diff
@@ -130,21 +131,26 @@
...
-			icon_state = "holo_medical-deny"
-			return FALSE
-		else
-			return TRUE //nice or benign diseases!
+		return CheckHuman(mover)
+
+/obj/structure/holosign/barrier/medical/Bumped(atom/movable/AM)
+	. = ..()
+	icon_state = "holo_medical"
+	if(ishuman(AM) && !CheckHuman(AM))
```
Comment (optimumtact) [inline code/game/objects/structures/holosign.dm]: what in gods name is this

Comment (optimumtact) [reply code/game/objects/structures/holosign.dm]: oh it's right below nvm

**Comment by ninjanomnom** [comment]
I've heard no complaints from the testmerge

**Pull request merged**